<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1053</width>
    <height>498</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenGLDemo</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="MyGL" name="mygl">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>12</y>
      <width>511</width>
      <height>431</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="vertsListWidget">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>10</y>
      <width>91</width>
      <height>161</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="halfEdgesListWidget">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>10</y>
      <width>101</width>
      <height>161</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="facesListWidget">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>10</y>
      <width>101</width>
      <height>161</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>180</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Vertices</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>180</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Half-Edges</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>180</y>
      <width>81</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Faces</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="vertPosXSpinBox">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>210</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.989999999999995</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="vertPosYSpinBox">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>210</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.989999999999995</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="vertPosZSpinBox">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>210</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.989999999999995</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>210</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Vertex Position</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>230</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>230</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>230</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>310</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Blue</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>310</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Red</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="faceBlueSpinBox">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>290</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.050000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>290</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Face Color</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="faceGreenSpinBox">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>290</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.050000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="faceRedSpinBox">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>290</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.050000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>310</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Green</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="addMidpoint">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>340</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Add Midpoint</string>
    </property>
   </widget>
   <widget class="QPushButton" name="triangulate">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>340</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Triangulate</string>
    </property>
   </widget>
   <widget class="QPushButton" name="subdivide">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>340</y>
      <width>80</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Subdivide</string>
    </property>
   </widget>
   <widget class="QPushButton" name="load">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>370</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Load OBJ</string>
    </property>
   </widget>
   <widget class="QPushButton" name="json">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>370</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Load JSON</string>
    </property>
   </widget>
   <widget class="QTreeWidget" name="treeWidget">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>10</y>
      <width>181</width>
      <height>391</height>
     </rect>
    </property>
    <column>
     <property name="text">
      <string notr="true">1</string>
     </property>
    </column>
   </widget>
   <widget class="QPushButton" name="skin">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>370</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Skin</string>
    </property>
   </widget>
   <widget class="QPushButton" name="rotateX">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>400</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotate X</string>
    </property>
   </widget>
   <widget class="QPushButton" name="rotateY">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>400</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotate Y</string>
    </property>
   </widget>
   <widget class="QPushButton" name="rotateZ">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>400</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotate Z</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="jointXPos">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>250</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-10.000000000000000</double>
    </property>
    <property name="maximum">
     <double>10.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="jointYPos">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>250</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-10.000000000000000</double>
    </property>
    <property name="maximum">
     <double>10.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="jointZPos">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>250</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-10.000000000000000</double>
    </property>
    <property name="maximum">
     <double>10.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>250</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Joint Position</string>
    </property>
   </widget>
   <widget class="QLabel" name="wDisplay">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>440</y>
      <width>41</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>W</string>
    </property>
   </widget>
   <widget class="QLabel" name="iDisplay">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>440</y>
      <width>41</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>i</string>
    </property>
   </widget>
   <widget class="QLabel" name="jDisplay">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>440</y>
      <width>47</width>
      <height>14</height>
     </rect>
    </property>
    <property name="text">
     <string>j</string>
    </property>
   </widget>
   <widget class="QLabel" name="kDisplay">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>440</y>
      <width>47</width>
      <height>14</height>
     </rect>
    </property>
    <property name="text">
     <string>k</string>
    </property>
   </widget>
   <widget class="QLabel" name="xDisplay">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>440</y>
      <width>47</width>
      <height>14</height>
     </rect>
    </property>
    <property name="text">
     <string>x</string>
    </property>
   </widget>
   <widget class="QLabel" name="yDisplay">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>440</y>
      <width>47</width>
      <height>14</height>
     </rect>
    </property>
    <property name="text">
     <string>y</string>
    </property>
   </widget>
   <widget class="QLabel" name="zDisplay">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>440</y>
      <width>47</width>
      <height>14</height>
     </rect>
    </property>
    <property name="text">
     <string>z</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1053</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionCamera_Controls"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionCamera_Controls">
   <property name="text">
    <string>Camera Controls</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGL</class>
   <extends>QOpenGLWidget</extends>
   <header>mygl.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
